apps/smart-contracts/token/test/Vesting.test.ts

```diff
@@ -1054,4 +1067,216 @@ describe('Vesting', () => {
       )
     })
   })
+
+  describe(' # testEndToEnd', () => {
+    let ppo: PPO
+    let sourceAllowlist: AccountList
+    let destinationAllowlist: AccountList
+    let blocklist: AccountList
+    let restrictedTransferHook: RestrictedTransferHook
+    let allowedSourceUser: SignerWithAddress
+    let allowedDestinationUser: SignerWithAddress
+    let blockedUser: SignerWithAddress
+    let unblockedUser1: SignerWithAddress
+    let unblockedUser2: SignerWithAddress
+    let disallowedSourceUser: SignerWithAddress
+    let disallowedDestinationUser: SignerWithAddress
+    let blockedRecipient: SignerWithAddress
+
+    const getSignersForEndToEndStack = async (): Promise<void> => {
+      ;[allowedSourceUser, allowedDestinationUser, blockedUser, unblockedUser1, unblockedUser2] =
+        await ethers.getSigners()
+    }
+
+    const ONE_PPO = parseEther('1')
+    const VESTING_PPO_SUPPLY = ONE_PPO.mul(100)
+
+    const setupEndToEndStack = async (): Promise<void> => {
+      await getSignersForEndToEndStack()
+      // Setup Vesting
+      await deployVesting()
+      await vesting.connect(owner).acceptOwnership()
+      // Setup PPO
+      ppo = await ppoFixture('prePO Token', 'PPO', owner.address)
+      await ppo.connect(owner).acceptOwnership()
+      // Set Vesting token as PPO
+      await vesting.connect(owner).setToken(ppo.address)
+      // Setup Source Allowlist
```

Comment:
```suggestion
      // Setup source allowlist
```
This is just the contextual name of this entity, not the name of the contract itself, can leave lowercase.